From 4844facff6b6ae66d7d1d9abd3c45bd470d5dfdc Mon Sep 17 00:00:00 2001
From: Daniel-Constantin Mierla <miconda@gmail.com>
Date: Wed, 15 Feb 2017 12:53:59 +0100
Subject: [PATCH] sanity: reset proxy_require hdr parsed field after freeing

(cherry picked from commit 06e219158131c2397a96bf5d3373c96752d7fd20)
(cherry picked from commit 51949e50e0308c9ec41cd95bfe2be3d64e24802d)
---
 modules/sanity/sanity.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/sanity/sanity.c b/modules/sanity/sanity.c
index 8140c07b2..581219001 100644
--- a/modules/sanity/sanity.c
+++ b/modules/sanity/sanity.c
@@ -724,6 +724,7 @@ int check_proxy_require(struct sip_msg* _msg) {
 			 * freed when the message freed. Lets hope nobody needs to access
 			 * this header again later on */
 			free_str_list(_msg->proxy_require->parsed);
+			_msg->proxy_require->parsed = NULL;
 		}
 	}
 #ifdef EXTRA_DEBUG
-- 
2.11.0

